### DISCUSSION TITLE
[Challenge, 25USDT/Tether] I need help with a simple regex / scrapping of SQL tables

### DISCUSSION THREAD

**Discussion by seychelles111** [antlr/antlr4#4239] [Q&A]
**can you please help me? Ever liked CTF's or any challenges 

 

see it as a chess challenge, just wheres is someone failed before you,

I always like to help people when ever I can, even if its buying Pepsi/Beer to an begger-person - instead of giving him/her/them money for drugs! 

I am in need of help, sometimes you need to ask for help** 

If anyone decides to help, me I want to tip back, 
as I think its appropriate, dont forget to mention the crypto-type of your liking :)  
---------------
example of parsing:

Something I tried but failed
https://gist.github.com/seychelles111/618a8b1aa0b77ab1289543a883811d18
**due to different "," placements**
![image](https://user-images.githubusercontent.com/130751272/232776390-9407afdb-66f1-43bf-b0e2-afe11d8c81f7.png)

-------------------------------------------------------------------------------

example of input text 
https://gist.github.com/seychelles111/a831cf4ab2e4340ce8e8488085ea31de

irefrence:
SQL*Loader DDL Syntax Diagram Notation https://docs.oracle.com/cd/F49540_01/DOC/server.815/a67792/ch05.htm#1462
[Raw](https://gist.github.com/seychelles111/a831cf4ab2e4340ce8e8488085ea31de/raw/d369390fe08141b3adf7a060f2739da04d1389b3/plsql_sql.ctl)

 file - 1 
https://www.oracle.com/webfolder/technetwork/docs/fbdi-23a/fbdi/controlfiles/PoDistributionsInterfaceOrder.ctl
 
 file - 2 
https://www.oracle.com/webfolder/technetwork/docs/fbdi-23a/fbdi/controlfiles/DooOrderChargeTiersInt.ctl

 file - 3
https://www.oracle.com/webfolder/technetwork/docs/fbdi-23a/fbdi/controlfiles/DooOrderChargeTiersInt.ctl

------

I cant scrap more than one table   I been stack on it for hours, I have no ideas what to do, 

if you like challenges, and alright with regex, [[Sound-effect: – `Dark Vader's challenge accepted`]]

can you please help me? Ever liked CTF's or any challenges

**Comment by seychelles111** [comment]
I am willing to give 
25USDT, or 50$ for charity of your liking. in return for help :)

**Comment by seychelles111** [comment]
Hey @raffian @KvanTTT @jimidle @aphyr @tonyarnold

please check this bounty :)

**Comment by seychelles111** [comment]
@sharwell 
are you willing to help a noob like me ?

**Comment by jimidle** [comment]
This might by an XY question. 

Can you say what you are trying to accomplish that makes you feel that you need to parse these load controllers?

Also, this is Oracle?

**Reply by seychelles111** [reply]
Hey this is Oracle SQLLDR, 
Its based on the DDL
https://docs.oracle.com/cd/B14117_01/server.101/b10759/statements_1001.htm#i2099120
also:
https://docs.oracle.com/database/121/SUTIL/GUID-AEBA1A45-2394-4947-81CB-37CD99BA5BA9.htm

I am really hopeless, I dont know how to deal with it

some times the "," comma is on the left side and sometimes not.. sometimes there is a \t 

I am confused

**Reply by seychelles111** [reply]
Also thanks for giving me the time and answering

**Reply by seychelles111** [reply]
> This might by an XY question.
> 
> Can you say what you are trying to accomplish that makes you feel that you need to parse these load controllers?
> 
> Also, this is Oracle?

I need to load the loaders because I am trying to help some people, who are having issues with something. 
I am not a developer, alltho I like to build thing and be a part of something that people use..

I need the scrap that, so I could have an excel, and see where values apear somewhere else, 
I've made an automated script/python-code that auto scraps those fields and compare everything (I rather use Python, but I am trying to help people who do BI) I am trying to make a simple python script that compare each table and each and where it apears in the documentation 


![image](https://user-images.githubusercontent.com/130751272/232850632-96ca780e-5ad6-45d3-b0ab-6e85f2b0c77a.png)

![image](https://user-images.githubusercontent.com/130751272/232850695-b3e34df0-fac5-4b92-af34-cf5f0c4d6507.png)

https://docs.oracle.com/en/cloud/saas/supply-chain-management/23a/oefsc/importmeterreadings-3113.html#importmeterreadings-3113

I have no idea how to do it :( 
Im trying to help other people too actually, they really try to do it manually,  I think its time that could be used to read the docs and undersrand the software better. 

![image](https://user-images.githubusercontent.com/130751272/232851483-dd552453-e660-41ed-aa48-0c4c6396193c.png)
